データベース設計

nemoimoi opened the pull request:
# What
ER図とREADMEを作成しDB設計

# Why
実装に必要なテーブルとカラム情報洗い出すため。

# ER図
https://gyazo.com/dd63a770ab8f60eb5abb92101eb0fb12

nemoimoi committed:
データベース設計

motomuun commented on README.md:
```diff
@@ -1,24 +1,59 @@
+|----------------|----------|-----------------|
+|nickname        |string    |null: false      |
+|email           |string    |null: false,unique: true|
+|password        |string    |null: false      |
```
`|password        |string    |null: false      |`
こちらのカラム名はdeviseで用意されている`encrypted_password`を使用しましょう！

motomuun commented on README.md:
```diff
@@ -1,24 +1,59 @@
+|first_name_kana |string    |null: false      |
+|birth_yyyy      |integer   |null: false      |
+|birth_mm        |integer   |null: false      |
+|birth_dd        |integer   |null: false      |
```
```
|birth_yyyy      |integer   |null: false      |
|birth_mm        |integer   |null: false      |
|birth_dd        |integer   |null: false      |
```
生年月日は配布素材のビューファイルに従った仕様で実装する必要がございますので、
一つのカラムにまとめましょう！

また、生年月日のデータ型は`date`を指定しましょう！

motomuun commented on README.md:
```diff
@@ -1,24 +1,59 @@
+|title           |string    |null: false      |
+|description     |text      |null: false      |
+|content         |text      |null: false      |
+|about_shipping  |string    |null: false      |
```
プルダウンで選択する項目のカラムが不足しているようです。
必要なカラムを洗い出し、active_hashを用いて実装しましょう！

active_hashでの実装に際し、以下についてご確認をお願いいたします！

①
active_hashを使用して保存するデータは数値を扱うので、データ型はintegerを指定しましょう！

②
active_hashを使用して保存するデータのカラムは、「単数系_id」という形にする必要がございます。
以下のように各カラム名を修正しましょう！

```
例：category_id
```
**********

motomuun commented on README.md:
```diff
@@ -1,24 +1,59 @@
+
+### Association
+- belongs_to :user
+- has_one :shipping_information
```
`has_one :shipping_information`
商品テーブルと結びつくのは配送先住所ではなく購入履歴テーブルです。
アソシエーションを結ぶテーブル名を修正しましょう。

motomuun commented on README.md:
```diff
@@ -1,24 +1,59 @@
+## paymentsテーブル
+|Column          |Type      |Options          |
+|----------------|----------|-----------------|
+|buy             |string    |null: false	    |
```
`|buy             |string    |null: false	    |`

購入履歴テーブルに保存するデータは、「どのユーザー」が「どの商品」を購入したかです。
「どのユーザー」かは指定できておりますので、
「どの商品」かを商品情報テーブルの外部キーとして、カラムを指定しなおしましょう！

motomuun commented on README.md:
```diff
@@ -1,24 +1,59 @@
+- belongs_to :item
+- has_one :shipping_information
+
+## shipping_informationテーブル
```
`## shipping_informationテーブル`
**テーブル名は複数形で命名しましょう！**
命名規則として、テーブル名は複数形で名付ける必要があるためです。

motomuun commented on README.md:
```diff
@@ -1,24 +1,59 @@
+## shipping_informationテーブル
+|Column          |Type      |Options          |
+|----------------|----------|-----------------|
+|postcode        |integer   |null: false	    |
```
`|postcode        |integer   |null: false	    |`
郵便番号には「-」を含める必要があるため、string型を指定しましょう！

motomuun commented on README.md:
```diff
@@ -1,24 +1,59 @@
+|municipality    |string    |null: false	    |
+|house_number    |string    |null: false	    |
+|building_name   |string    |null: false	    |
+|telephone_number|integer   |null: false	    |
```
`|telephone_number|integer   |null: false	    |`
電話番号の多くは0から始まりますが、integer型は0から始まる整数を識別できません。
そのため、電話番号はstring型で指定しましょう！

motomuun commented on README.md:
```diff
@@ -1,24 +1,59 @@
+|prefecture      |string    |null: false	    |
+|municipality    |string    |null: false	    |
+|house_number    |string    |null: false	    |
+|building_name   |string    |null: false	    |
```
`|building_name   |string    |null: false	    |`
建物名の入力は任意のため、null: falseは不要でございます。

motomuun commented on README.md:
```diff
@@ -1,24 +1,59 @@
+|house_number    |string    |null: false	    |
+|building_name   |string    |null: false	    |
+|telephone_number|integer   |null: false	    |
+|user            |references|null: false,foreign_key: true|
```
`|user            |references|null: false,foreign_key: true|`
userではなく紐づいている購入履歴のレコードのidを保存するカラムが必要です。
購入履歴テーブルの外部キーとしてカラムを追加しましょう！
また、usersテーブルの外部キーは削除しましょう。

motomuun commented on README.md:
```diff
@@ -1,24 +1,59 @@
+|Column          |Type      |Options          |
+|----------------|----------|-----------------|
+|postcode        |integer   |null: false	    |
+|prefecture      |string    |null: false	    |
```
`|prefecture      |string    |null: false	    |`
こちらはプルダウン形式のためactive_hashを使用します。
また、商品情報テーブルで使用したactive_hashのデータを兼用できます。
商品情報で指定した後、同じものをこちらでも指定しましょう！

nemoimoi replied:
レビューありがとうございます！
コメント頂いた部分を修正&変更いたしました！
再度レビューのほどよろしくお願いいたします。

motomuun reviewed (commented):
レビュー依頼ありがとうございます！
カラムの命名規則が適切かつ明瞭にできており素晴らしいです👏

修正点をコメントさせて頂きましたので、ご確認をお願い致します。
ご不明点や疑問点がありましたら、チャット・コネクトメンターにご相談ください。

【次回プルリクエスト提出する際の注意点】
①修正が終わりましたら、`コミットとプッシュ`をして、`プルリクエストに修正内容が反映されているか確認`しましょう。
②最新のER図の提出をお願い致します。
③googleフォームの送信をお願いいたします。

nemoimoi committed:
データベース設計デバック

・パスワードカラム名変更
・生年月日カラムを集約
・itemsテーブルカラム名をActivehashを使用する想定の内容に修正&追記
・itemsテーブルとordersテーブルのアソシエーションを変更
・商品発送テーブル名を変更
・購入テーブル名を変更
・郵便番号のデータ型をstring型に変更
・電話番号のデータ型をstring型に変更
・建物名のオプションを任意入力に変更
・orderテーブルにitemカラムを追加。
・usersテーブル外部キーを削除
・発送先の都道府県はactivw_hashを用いるカラム名に変更

nemoimoi reviewed (commented):
# What
データベース設計のレビューをいただいたのでコメントを元にデバックしました
# Why
DB設計後実装に必要なカラムの追記やカラム・テーブル名の変更が必要だったため

# ER図
https://gyazo.com/63979eda0fa6fa2c35ac5b7bd88edb1d

shodai-ito commented on README.md:
```diff
@@ -1,24 +1,61 @@
+## ordersテーブル
+|Column          |Type      |Options          |
+|----------------|----------|-----------------|
+|buy             |string    |null: false	    |
```
`|buy             |string    |null: false	    |`
こちらのカラムは不要ですので、削除しましょう！

shodai-ito commented on README.md:
```diff
@@ -1,24 +1,61 @@
+|house_number    |string    |null: false	    |
+|building_name   |string    |           	    |
+|telephone_number|string    |null: false	    |
+|item            |references|null: false,foreign_key: true|
```
`|item            |references|null: false,foreign_key: true|`
住所テーブルがアソシエーションを組んでいるのはorderですので、カラム名はorderが適切かと存じます。

shodai-ito reviewed (commented):
レビュー依頼ありがとうございます！
プルリクエストの内容を確認させていただきました。

修正点をコメントさせて頂きましたので、ご確認をお願い致します。

【次回プルリクエスト提出する際の注意点】
```ruby
①修正が終わりましたら、`コミットとプッシュ`をして、`プルリクエストに修正内容が反映されているか確認`しましょう。
②既に頂いている挙動確認の画像や動画が修正範囲だった場合、最新の画像や動画の提出をお願い致します。
③googleフォームの送信をお願いいたします。
```


ご不明点や疑問点ございましたら、チャット・コネクトメンターにご相談ください！
どうぞよろしくおねがいします。

nemoimoi committed:
データベース設計デバック２

・不要カラムの削除
・アソシエーションのカラム名変更

nemoimoi committed:
データベース設計デバック３

・README記述追加

nemoimoi commented:
レビューありがとうございます！
コメント頂いた箇所を改めて変更いたしました！

nemoimoi reviewed (commented):
# What
ER図とREADMEの修正を行いました

# Why
不要なカラムの記載と異なるテーブルのカラムが記述されていたため修正を行いました

# ER図
https://gyazo.com/6d20c7f62b64e0ccf77853b2812c782c

TakashiKozuma commented:
追加のレビュー依頼ありがとうございます！
修正点も含め、確認が完了しました。
LGTMです！！

こちらのブランチは忘れずにMergeいただき、ローカルのMasterにPullした後に次の実装のためのブランチを切ってから次にお進み下さい。

nemoimoi commented:
LGTMありがとうございます！
MergeしてPullを行い次のブランチ作成してユーザー管理機能の実装に進みます！
